### PR TITLE
Add the ability to enable debug logs in the config entry overflow

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Add the ability to enable debug logs in the config entry overflow

~~Needs https://github.com/home-assistant/core/pull/65158~~

After a few months, I realized I was overthinking this and if they want to turn on debug logging for a YAML only integration its reasonable to expect that if they can write YAML they can manually add `logger` YAML.   https://ptb.discord.com/channels/330944238910963714/427516175237382144/1037451384548765779

debug_logging

demo2

A future improvement might be to also add a button to download the full log

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bdraco** [comment]
Finished re-test. Should be ready for review now

**Comment by balloob** [comment]
So I had a different idea for this. When we enable debug logging, we still make it the users problem to find them.

Instead, what if we enable a diagnostic mode, and when stopped (maybe default after 60s), it downloads a file that has the logging for that integration + any extra things the integration might want to log (ie events).

**Comment by bdraco** [comment]
> So I had a different idea for this. When we enable debug logging, we still make it the users problem to find them.
> 
> Instead, what if we enable a diagnostic mode, and when stopped (maybe default after 60s), it downloads a file that has the logging for that integration + any extra things the integration might want to log (ie events).

I think that could be accomplished by automatically downloading the log file the call added in https://github.com/home-assistant/frontend/pull/14404 after you turn off debug logging. 

I'm not sure if thats 100% what you had in mind though. Alternately, how would you envision enabling/disabling diagnostics mode?

**Comment by bdraco** [comment]
Another thing we could do is change diagnostics to be a zip file which includes the log but I'd worry about people uploading zip files before checking the log for anything sensitive.

**Comment by balloob** [comment]
Oh it is a nice idea to offer people the option to download the logs when they turn off the logging.

**Comment by bdraco** [comment]
Will come back to this after the holiday

**Comment by bdraco** [comment]
I can use the new code in https://github.com/home-assistant/frontend/pull/14404 to have it download once that merges

**Comment by bdraco** [comment]
The debug log now downloads when you disable debug logging

**Comment by bdraco** [comment]
Retested. This should be ready to go.

**Review by bramkragten — APPROVED** [review]

**Comment by mib1185** [comment]
> A future improvement might be to also add a button to download the full log

- Sounds well known https://github.com/home-assistant/frontend/pull/11427 🙈

File: src/data/integration.ts [hunk 1]
```diff
@@ -36,6 +39,24 @@ export interface IntegrationSetup {
...
 }
 
+export interface IntegrationLogInfo {
+  domain: string;
+  level?: number;
+}
+
+export enum LogSeverity {
+  CRITICAL = 50,
+  FATAL = 50,
```
Comment (steverep) [inline src/data/integration.ts:49]: Why 2 names with the same value?  New rules in #17299 throw an error for this, and doesn't look like we use either.

Comment (bdraco) [reply src/data/integration.ts:49]: [Python changed the name but preserved backcompat](https://github.com/python/cpython/blob/263d8aa017bd6d858aa9488154bc160fa7249a9e/Lib/logging/__init__.py#L92)

Comment (steverep) [reply src/data/integration.ts:49]: But there's no reason we need to keep fatal in the `enum`, right?

Comment (bdraco) [reply src/data/integration.ts:49]: I think its fine to drop it

**Pull request merged**